### PR TITLE
[#116] Add RegistryDao decision lambda test

### LINKED ISSUE #116: Implement Registry DAO using LIGO version
# Clarification and motivation

Now that we have a LIGO version of BaseDAO we want to demonstrate its usage. Let's implement [RegistryDAO](https://github.com/tqtezos/baseDAO/blob/master/docs/registry.md) as an example.

# Acceptance criteria

1. Lambdas that are used to parameterize BaseDAO are implemented (in LIGO) according to the RegistryDAO specification.
2. There are instructions how to originate LIGO version of RegistryDAO and these instructions are as simple as possible.

### PR DESCRIPTION
**Pull request by sras**
Problem: The tests for decision lambda are missing, because it is hard
to check for storage changes since it contains bigmaps.

Solution: Implement a test for decision lambda behavior where we update
s_max value with a proposal, and test it by creating another proposal
which should only work if the `s_max` value was updated.

## Related issue(s)

Resolves #116

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [ ] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [ ] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [ ] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: ligo/haskell/test/Test/Ligo/RegistryDAO.hs [hunk 1]
```diff
@@ -128,11 +129,62 @@ test_RegistryDAO =
 
               let spent = div (requiredFrozen * c) d
 
-              checkStorage (AddressResolved $ unTAddress consumer) (toVal [[FA2.BalanceResponseItem balanceRequestItem (100 - spent)]])
+              checkStorage (AddressResolved $ unTAddress consumer) (toVal [[FA2.BalanceResponseItem balanceRequestItem (defaultTokenBalance - spent)]])
+
+    , nettestScenarioCaps "checks it correctly calculates executes proposal that has won" $ do
```
Comment (pasqu4le) [inline ligo/haskell/test/Test/Ligo/RegistryDAO.hs:134]:
```suggestion
    , nettestScenarioCaps "checks it correctly executes the proposal that has won" $ do
```

File: ligo/haskell/test/Test/Ligo/RegistryDAO.hs [hunk 2]
```diff
@@ -128,11 +129,62 @@ test_RegistryDAO =
...
+                    ,([mt|b|], lPackValueRaw @(Maybe Natural) Nothing)
+                    ,([mt|c|], lPackValueRaw @(Maybe Natural) Nothing)
+                    ,([mt|d|], lPackValueRaw @(Maybe Natural) Nothing)
+                    ]
+              let sMaxUpdateproposalSize1 = metadataSize sMaxUpdateproposalMeta1
+              let requiredFrozenForUpdate = sMaxUpdateproposalSize1 * a + b
+
+              callFrom (AddressResolved wallet1) baseDao (Call @"Propose") (ProposeParams requiredFrozenForUpdate sMaxUpdateproposalMeta1)
+
+              -- Then we sent 2 upvotes for the proposal (as max quorum is 2)
```
Comment (pasqu4le) [inline ligo/haskell/test/Test/Ligo/RegistryDAO.hs:173]:
```suggestion
              -- Then we send 2 upvotes for the proposal (as min quorum is 2)
```

**Review by pasqu4le — COMMENTED** [review]
Nitpick, but in the commit (and PR) description it says:
> Problem: The tests for decision lambda are missing, because it is hard
to check the storage changes since it contains bigmaps.

I think `to check the storage changes` should be replaced with `to check for storage changes` or `to check that the storage changes`.
Maybe it's fine either way, but I think either of these is easier to read.

**Review by pasqu4le — APPROVED** [review]
LGTM, if possible I'd like @rinn7e to review this as well

**Review by rinn7e — APPROVED** [review]
Seems good for me!

**Pull request merged**